src/convertToObject.js

```diff
@@ -3,16 +3,35 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example in
+ * [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties (see an exampl
+ * in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const properties = sourceString
+    .split(';')
+    .map(prop => prop.split(':'));
+
+  const propertiesWithValues = properties
+    .filter(prop => prop.length > 1);
+
+  const clearProperties = propertiesWithValues
+    .map(prop => ([prop[0].trim(), prop[1].trim()]));
```

Comment:
Не впевнений, що в цьому випадку потрібно створювати змінні, наврядчи десь буде потрібно використовувати ці проміжні обробки для отримання `properties`. Можливо було б краще зробити їх одним ланцюжком.

```suggestion
  const properties = sourceString
    .split(';')
    .map(prop => prop.split(':'))
    .filter(prop => prop.length > 1)
    .map(prop => ([prop[0].trim(), prop[1].trim()]));
```

Reply:
згоден. в мене чогось тести до цього не проходили. і я нуплужив такого коду)))